main.tf

```diff
@@ -1,6 +1,6 @@
 locals {
   master_password      = var.create_db_instance && var.create_random_password ? random_password.master_password[0].result : var.password
-  db_subnet_group_name = var.replicate_source_db != null ? null : coalesce(var.db_subnet_group_name, module.db_subnet_group.db_subnet_group_id)
+  db_subnet_group_name = var.replicate_source_db != null || var.create_db_subnet_group == false ? null : coalesce(var.db_subnet_group_name, module.db_subnet_group.db_subnet_group_id)
```

Comment:
Change this value to this:
```
db_subnet_group_name = var.replicate_source_db != null ? null : var.create_db_subnet_group == false ? null : coalesce(var.db_subnet_group_name, module.db_subnet_group.db_subnet_group_id)
```
This works in honoring new logic and old flag logic. Testing now.